gateways/adyen/request_builder.go

```diff
@@ -92,7 +93,17 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest, merchantAccount s
 	}
 
 	level3 := authRequest.Level3Data
-	if level3 != nil {
+	if authRequest.Options[googlePayTokenOption] != nil && level3 == nil {
+		googlePayData := map[string]interface{}{
+			"payment.token": authRequest.Options[googlePayTokenOption].(string),
+		}
+		request.AdditionalData = googlePayData
+	} else if level3 != nil && authRequest.Options[googlePayTokenOption] != nil {
+		var builtL3Data map[string]string
+		builtL3Data = buildLevel3Data(level3)
+		builtL3Data["payment.token"] = authRequest.Options[googlePayTokenOption].(string)
+		request.AdditionalData = builtL3Data
+	} else if level3 != nil {
```

Comment:
nit: we could probably simplify this a bit?
```suggestion
	if authRequest.Options[googlePayTokenOption] != nil {
		googlePayData := map[string]interface{}{
			"payment.token": authRequest.Options[googlePayTokenOption].(string),
		}
		request.AdditionalData = googlePayData
	} 
	if level3 != nil  {
	        if request.AdditionalData == nil {
	            request.AdditionalData = make(map[string]string)
	        }
		buildLevel3Data(level3, request.AdditionalData)
	} 
```

Reply:
actually getting rid of this as its the wrong approach